### ISSUE TITLE
Asciidoctor internal cross references and fonts in general

### ISSUE THREAD

**Issue by wehimwich** [nvi-inc/fsl10#17]
It looks to me like there is an incompatibility between how _Asciidoctor_ 1.5.4 (FSL10) and  2.0.10 (Mac) handle manually written internal cross-references. For example the links in the first two paragraphs at

https://nvi-inc.github.io/fsl10/#_managing_security_updates

don't work properly with 1.5.4 (which is what is used for the above document), but do with 2.0.10. This appears to be related to use of special fonts in the section titles being referred to. Automatically generated links (e.g., in the TOC) work fine, but the format appears to be different between the two versions (e.g., _strong_fsadapt_strong in 1.5.4 and _fsadapt in 2.0.10). Maybe the best solution is to remove all special fonts from section titles.

Also, I am starting to regret my decision to use bold for so much: inline code, filenames, programs names, and empahsis. I picked bold because I thought (and still think) that the inline monospace with its different background did not look very good and bold gave better visibility for emphasis. However, maybe I am swimming against too strong a tide. We could go back to more O'Reilly-like font conventions, which I think with _Asciidoctor_ might be approximated as :

- _italics_ is used for program names, emphasis, and definition of technical terms
- `monospace` is used for file names, code blocks, program output
-  **`bold monospace`**  for user input/commands, inline only (not in code blocks)
- _`monospace italic`_ for replaceable items for inline (only) input/commands
- and: no special fonts in  the section titles due to the _Asciidoctor_ version issues.

Please let me know if I didn't get that quite right.

Any thoughts or suggestions?

**Comment by dehorsley** [comment]
> Maybe the best solution is to remove all special fonts from section
titles.

Sounds good, they're probably more of a distraction than anything anyway.

> Also, I am starting to regret my decision to use bold for so much: inline
code, filenames, programs names, and empahsis. I picked bold because I
thought (and still think) that the inline monospace with its different
background did not look very good and bold gave better visibility for
emphasis.

Yeah, I find the excess of bold face distracting; it's somewhat too
eye-catching. I don't mind the different background colour on the monospace
so much, but we can change the style sheet to remove it.

> We could go back to more O'Reilly-like font conventions

Sounds good to me.

**Comment by wehimwich** [comment]
>Yeah, I find the excess of bold face distracting; it's somewhat too
eye-catching. I don't mind the different background colour on the monospace
so much, but we can change the style sheet to remove it.

Well I'm an old guy; it takes more and more to catch my attention. I can probably adapt to the background. It is eye catching in a lower key way.

**Comment by jfhquick** [comment]
On Sun, June 14, 2020 10:25 pm, Ed Himwich wrote:
> Also, I am starting to regret my decision to use bold for so much: inline
> code, filenames, programs names, and empahsis. I picked bold because I thought
> (and still think) that the inline monospace with its different background did
> not look very good and bold gave better visibility for emphasis. However,
> maybe I am swimming against too strong a tide. We could go back to more
> O'Reilly-like font conventions, which I think with _Asciidoctor_ might be
> approximated as :
>
> - _italics_ is used for program names, emphasis, and definition of technical
> terms
> - `monospace` is used for file names, code blocks, program output
> -  **`bold monospace`**  for user input/commands, inline only (not in code
> blocks)
> - _`monospace italic`_ for replaceable items for inline (only) input/commands
> - and: no special fonts in  the section titles due to the _Asciidoctor_
> version issues.

Sounds good, but will take some doing to get it all consistent :-S

J.

**Comment by wehimwich** [comment]
>Sounds good, but will take some doing to get it all consistent :-S

Yeah something to look at some day I guess. Maybe cleaning up the bold in inline text would be relatively easy and give the most benefit.

I did find a way to use **`bold`** and _`italic`_ (and even _**`bold italic`**_ ) in a code block with a construction like:

```
[source,bash,subs="verbatim,quotes"]
----
your code here
----
```

And it works for both Asciidoctor versions, joy of joys. So if we want to be even more O'Reilly-like, it could be:

 - _italics_ is used for program names, emphasis, email address, URLs, domain names, nodes, and first use of technical terms
 - `monospace` is used for file names, code, program output
 -  **`monospace bold`**  for user input/commands
 - _`monospace italic`_ for replaceable items for input/commands
 - and: no special fonts in  the section titles due to the _Asciidoctor_ version issues.

The whole O'Reilly conventions are at:

http://oreillymedia.github.io/production-resources/styleguide/#typography_and_font_conventions

and how to do it in Asciidoc:

https://docs.atlas.oreilly.com/writing_in_asciidoc.html#INLINES

I am really glad I ran this to ground. :(

**Comment by wehimwich** [comment]
I pushed a commit (c8394f50e5017fb72d087ff7a8abcefb7335a526) that accomplishes the first step below:

- [x] Remove special fonts from section titles (only bold and italics were in use)
- [x] Correct usage of bold (and others) outside of code blocks
- [x] Correct fonts in code blocks

**Comment by wehimwich** [comment]
This was effectively closed by 395d72a51b9e5a63bd4f5b87f7530321e6567476. The documents now use the [FS conventions](https://nvi-inc.github.io/fs/misc/font_conventions.html).
